internal/controller/rotators/aws_oidc_rotator.go

```diff
@@ -116,58 +116,53 @@ func (r *AWSOIDCRotator) GetPreRotationTime(ctx context.Context) (time.Time, err
 	return preRotationTime, nil
 }
 
+func populateSecretWithAwsIdentity(secret *corev1.Secret, awsIdentity *sts.AssumeRoleWithWebIdentityOutput, region string) {
+	updateExpirationSecretAnnotation(secret, *awsIdentity.Credentials.Expiration)
+	// For now have profile as default.
+	const defaultProfile = "default"
+	credsFile := awsCredentialsFile{awsCredentials{
+		profile:         defaultProfile,
+		accessKeyID:     aws.ToString(awsIdentity.Credentials.AccessKeyId),
+		secretAccessKey: aws.ToString(awsIdentity.Credentials.SecretAccessKey),
+		sessionToken:    aws.ToString(awsIdentity.Credentials.SessionToken),
+		region:          region,
+	}}
+	updateAWSCredentialsInSecret(secret, &credsFile)
+}
+
 // Rotate implements the retrieval and storage of AWS sts credentials.
 //
 // This implements [Rotator.Rotate].
 func (r *AWSOIDCRotator) Rotate(ctx context.Context, token string) error {
-	r.logger.Info("rotating AWS sts temporary credentials",
-		"namespace", r.backendSecurityPolicyNamespace,
-		"name", r.backendSecurityPolicyName)
+	bspNamespace := r.backendSecurityPolicyNamespace
+	bspName := r.backendSecurityPolicyName
+	secretName := GetBSPSecretName(bspName)
 
-	result, err := r.assumeRoleWithToken(ctx, token)
+	r.logger.Info("rotating aws credentials secret", "namespace", bspNamespace, "name", bspName)
+	awsIdentity, err := r.assumeRoleWithToken(ctx, token)
 	if err != nil {
 		r.logger.Error(err, "failed to assume role", "role", r.roleArn, "access token", token)
 		return err
 	}
 
-	secret, err := LookupSecret(ctx, r.client, r.backendSecurityPolicyNamespace, GetBSPSecretName(r.backendSecurityPolicyName))
-	if err != nil {
-		if !apierrors.IsNotFound(err) {
-			return err
-		}
+	secret, err := LookupSecret(ctx, r.client, bspNamespace, secretName)
+
+	if err != nil && apierrors.IsNotFound(err) {
```

Comment:
Need to cover the case for errors other than `IsNotFound` which we can simply return the error.